### ISSUE TITLE
Regex has to be modified for ObjC block message syntax.

### ISSUE THREAD

**Issue by samishchandra** [uncrustify/uncrustify#2513]
Regex has to be modified for ObjC block message syntax. Note that without `id` in the block syntax works.

## # Input:
```
[something
 withBlock:^id (Component *c) {
      NSLog(@"Something");
    }];
```

## # ✅Expected Output:
```
[something
 withBlock:^id(Component *c) {
   NSLog(@"Something");
 }];
```

## # ❌Actual Output:
```
[something
 withBlock:^id (Component *c) {
  NSLog(@"Something");
}];
```

## # Config options used:
```
indent_columns = 2
indent_oc_block_msg = 2
indent_oc_msg_prioritize_first_colon = false
indent_oc_msg_colon = 0
align_oc_msg_spec_span = 0
align_oc_msg_colon_span = 0
align_oc_msg_colon_first = true
```

## # Uncrustify version
Uncrustify-369158f

**Comment by samishchandra** [comment]
We need to add the following lines in method `oc_msg_block_indent` in `indent.cpp`.

```
   if (chunk_is_paren_close(tmp))
   {
      tmp = chunk_get_prev_nc(chunk_skip_to_match_rev(tmp));
   }

   if (chunk_is_token(tmp, CT_TYPE))
   {
      tmp = chunk_get_prev_nc(tmp);
   }

   if (tmp == nullptr || tmp->type != CT_OC_BLOCK_CARET)
   {
      return(nullptr);
   }

```

Following is the new code that has to be added.
```
   if (chunk_is_token(tmp, CT_TYPE))
   {
      tmp = chunk_get_prev_nc(tmp);
   }
```

**Comment by mwoehlke-kitware** [comment]
I'm not sure I understand what you are proposing. Do you mean this?
```diff
    if (chunk_is_paren_close(tmp))
    {
       tmp = chunk_get_prev_nc(chunk_skip_to_match_rev(tmp));
    }

+   if (chunk_is_token(tmp, CT_TYPE))
+   {
+      tmp = chunk_get_prev_nc(tmp);
+   }
+
    if (tmp == nullptr || tmp->type != CT_OC_BLOCK_CARET)
    {
       return(nullptr);
    }
```

**Comment by samishchandra** [comment]
@mwoehlke yes that's correct. The existing logic is not taking the return type between `^` and `{` into account.

**Comment by gmaurel** [comment]
If you already have a proposal for this bug, please create a PR,
add some test data and comments.
Thanks.

**Comment by andrey-starodubtsev** [comment]
This issue is implicitly fixed after #2572 was merged.

**Comment by gmaurel** [comment]
fine
